tests/gold_tests/pluginTest/header_rewrite/header_rewrite_set_body_from.test.py

```diff
@@ -0,0 +1,156 @@
+        '''
+        Test where set-body-from request fails
+        Triggered from remap file
+        This uses the case where no remap rule is provided
+        '''
+        tr = Test.AddTestRun()
+        tr.Processes.Default.Command = (
+            'curl -s -v --proxy 127.0.0.1:{0} "http://www.example.com/remap_fail"'.format(self.ts.Variables.port))
+        tr.Processes.Default.ReturnCode = 0
+        tr.Processes.Default.StartBefore(self.server)
+        tr.Processes.Default.StartBefore(self.ts)
+        tr.Processes.Default.Streams.All = "gold/header_rewrite-set_body_from_remap_fail.gold"
+        tr.StillRunningAfter = self.server
+
+    def test_setBodyFromSucceeds_remap(self):
+        '''
+        Test where set-body-from request succeeds
+        Triggered from remap file
+        '''
+        tr = Test.AddTestRun()
+        tr.Processes.Default.Command = (
+            'curl -s -v --proxy 127.0.0.1:{0} "http://www.example.com/remap_success"'.format(self.ts.Variables.port))
+        tr.Processes.Default.ReturnCode = 0
+        tr.Processes.Default.Streams.All = "gold/header_rewrite-set_body_from_success.gold"
+        tr.StillRunningAfter = self.server
+
+    def test_setBodyFromSucceeds_plugin(self):
+        '''
+        Test where set-body-from request succeeds
+        Triggered from plugin file
+        '''
+        tr = Test.AddTestRun()
+        tr.Processes.Default.Command = (
+            'curl -s -v --proxy 127.0.0.1:{0} "http://www.example.com/plugin_success"'.format(self.ts.Variables.port))
+        tr.Processes.Default.ReturnCode = 0
+        tr.Processes.Default.Streams.All = "gold/header_rewrite-set_body_from_success.gold"
+        tr.StillRunningAfter = self.server
+
+    def test_setBodyFromFails_plugin(self):
+        '''
+        Test where set-body-from request fails
+        This uses the case where the second endpoint cannot connect to the requested server
+        Triggered from plugin file
+        '''
+        tr = Test.AddTestRun()
+        tr.Processes.Default.Command = (
+            'curl -s -v --proxy 127.0.0.1:{0} "http://www.example.com/plugin_fail"'.format(self.ts.Variables.port))
+        tr.Processes.Default.ReturnCode = 0
+        tr.Processes.Default.Streams.All = "gold/header_rewrite-set_body_from_conn_fail.gold"
+        tr.StillRunningAfter = self.server
+
+    def test_setBodyFromSucceeds_200(self):
+        '''
+        Test where set-body-from request succeeds and returns 200 OK
+        Triggered from remap file
+        This is tested because right now, TSHttpTxnErrorBodySet will change OK status codes to 500 INKApi Error
+        Ideally, this would not occur.
+        '''
+        tr = Test.AddTestRun()
+        tr.Processes.Default.Command = (
+            'curl -s -v --proxy 127.0.0.1:{0} "http://www.example.com/200"'.format(self.ts.Variables.port))
+        tr.Processes.Default.ReturnCode = 0
+        tr.Processes.Default.Streams.All = "gold/header_rewrite-set_body_from_200.gold"
```

Comment:
Testing against Streams.All seems to be making this test flaky/fragile.  I think you need your gold file to be matched against Streams.stdout, which I think would just contain the body.  You can test the response HTTP headers like this:  https://github.com/apache/trafficserver/blob/84206a8c089e56efc59fd08f2e3b693c413aaf03/tests/gold_tests/pluginTest/s3_auth/s3_auth_config.test.py#L60 .